jerry-ext/args/args.c

```diff
@@ -0,0 +1,97 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "jerryscript-ext/args.h"
+#include "jerryscript.h"
+
+/**
+ * It validate the JS arguments and assign them to the native argument.
+ *
+ * @return jerry undefined: the validator passes,
+ *         jerry error: the validator fails.
+ */
+jerry_value_t
+jerryx_validate_and_assign_args (const jerry_value_t *js_arg_p, /**< points to the js argument */
+                                 const jerry_length_t js_arg_cnt, /**< the count of the js arguments */
+                                 const jerryx_arg_t *c_arg_p, /**< points to the jerryx_arg_t */
+                                 const jerry_length_t c_arg_cnt) /**< the count of the native argument */
+{
+  jerry_value_t ret = jerry_create_undefined ();
+
+  if (c_arg_cnt == 0)
+  {
+    return ret;
+  }
+
+  jerryx_arg_iterator_t iterator =
```

Comment:
@akosthekiss Wow! Thanks for doing this deep dive.

I think you're proposal makes a lot of sense. I like it!

I think splitting off `c_arg_p` makes the API easier to understand. It's true to the original iterator pattern, whereas the current PR kind of mixes two iterators in one.. Sort of.

So we finally reached consensus on this one I guess?

Thanks again :)